### PR TITLE
chore(x-plugin): refactor and clean the code  

### PR DESCRIPTION
**Pull request by tajespasarela**
EX-4836

# Pull request template
This PR is only to refactor and clean the code. It doesn't change any feature or behaviour.

## Motivation and context
The motivation is to separate the emitters logic to add more complicated options for the emitters. The idea is to add more options or operators to the emitters in a future task.

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that causes existing functionality to not work as expected)
- [ ] Change requires a documentation update
- [x] Refactor code

## What is the destination branch of this PR?

- [x] `Main`
- [ ] Other. Specify:

## How has this been tested?
All tests and build checks must pass, as nothing change in the features or behaviour.

### CODE REVIEW

File: packages/x-components/src/plugins/x-emitters.ts [hunk 1]
```diff
@@ -0,0 +1,81 @@
+import { Store } from 'vuex';
+import { getGettersProxyFromModule } from '../store/utils/getters-proxy.utils';
+import { AnySimpleStateSelector, AnyStateSelector } from '../store/utils/store-emitters.utils';
+import { debounce } from '../utils/debounce';
```
Comment (pmareke) [inline packages/x-components/src/plugins/x-emitters.ts:4]: Remove unused import.

File: packages/x-components/src/plugins/x-emitters.ts [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+import { forEach } from '../utils/object';
+import { AnyXModule } from '../x-modules/x-modules.types';
+import { XBus } from './x-bus.types';
+
+/**
+ * Registers the store emitters, making them emit the event when the part of the state selected
+ * changes.
+ *
+ * @param xModule - The {@link XModule} to register its Store Emitters.
+ * @internal
```
Comment (pmareke) [inline packages/x-components/src/plugins/x-emitters.ts:14]: Remove duplicate internal

File: packages/x-components/src/plugins/x-plugin.ts [hunk 3]
```diff
@@ -236,29 +223,55 @@ export class XPlugin implements PluginObject<XPluginOptions> {
...
+      this.registerStoreEmitters(customizedXModule);
+      this.installedXModules.add(xModule.name);
     }
   }
 
+  /**
+   * Performs a customization of a {@link XModule} using the XPlugin public and private optioins.
+   *
+   * @param xModule - The module to customize.
+   * @returns The customized xModule.
```
Comment (pmareke) [inline packages/x-components/src/plugins/x-plugin.ts:240]: Add missing `@public`

**Review by pmareke — CHANGES_REQUESTED** [review]
Fix the build please.

**Review by CachedaCodes — CHANGES_REQUESTED** [review]

File: packages/x-components/src/plugins/x-emitters.ts [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+  });
+}
+
+/**
+ * Transforms a {@link AnySimpleStateSelector} into a {@link AnyStateSelector}, and sets
+ * default values for its properties.
+ *
+ * @param stateSelector - The state selector to normalize.
+ * @returns A {@link AnyStateSelector} with all the properties set.
+ * @internal
```
Comment (pmareke) [inline packages/x-components/src/plugins/x-emitters.ts:51]: I would add a blank line above as in the `registerStoreEmitters `.

File: packages/x-components/src/plugins/x-emitters.ts [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+  };
+}
+
+/**
+ * Checks if a the type of the store emitter selector is simple or complex. This selector can be
+ * a function if it is simple or an object with the selector and other options if it is complex.
+ *
+ * @param stateSelector - The store emitter selector.
+ * @returns A boolean which flags if the stateSelector is simple (function) or complex (object).
+ * @internal
```
Comment (pmareke) [inline packages/x-components/src/plugins/x-emitters.ts:73]: Same here.

File: packages/x-components/src/plugins/x-plugin.ts [hunk 6]
```diff
@@ -236,29 +223,56 @@ export class XPlugin implements PluginObject<XPluginOptions> {
...
+      this.installedXModules.add(xModule.name);
     }
   }
 
+  /**
+   * Performs a customization of a {@link XModule} using the XPlugin public and private options.
+   *
+   * @param xModule - The module to customize.
+   * @returns The customized xModule.
+   * @public
```
Comment (pmareke) [inline packages/x-components/src/plugins/x-plugin.ts:241]: Same here.

**Review by pmareke — CHANGES_REQUESTED** [review]
Some minor things.

**Review by pmareke — APPROVED** [review]

**Review by CachedaCodes — APPROVED** [review]

**Pull request merged**